SU2_CFD/src/solvers/CNEMOEulerSolver.cpp

```diff
@@ -912,6 +920,7 @@ void CNEMOEulerSolver::Upwind_Residual(CGeometry *geometry, CSolver **solver_con
         numerics->SetdTvedU(dTvedU_i, dTvedU_j);
         numerics->SetEve   (Eve_i,    Eve_j   );
         numerics->SetCvve  (Cvve_i,   Cvve_j  );
+        numerics->SetGamma (Gamma_i,  Gamma_i );
```

Comment:
I know we had a discussion about this and I might be forgetting something here. but it seems to me that this won't work properly. the way it is now:

Gamma_i, Gamma_j have not been updated with reconstructed variables, so they will have the same value associated to non-reconstructed variables nodes->GetGamma(iPoint)/ nodes->GetGamma(jPoint)

this is because we are not updating gamma in Cons2PrimVar. 

now, I am not saying to correct the situation, since this issue comes up because reconstruction has not been updated to Primitve variables (it still is in conservative variables).

should this be the time to include all the reconstruction in primitive vairables and make things consistent in this way?

if you do not want to mix up primitive reconstruction in this PR, can we temporarily have the gamma computation inside Cons2PrimVar for consistency ?

Reply:
That is correct,  the gamma passed in a MUSCL flow is not based on the reconstructed state.  

However to fix this, it is not so simple...
A gamma needs to be passed for the upwind residual to work properly -> the lazy fix would be to have a computeGamma() in the convective schemes. In this case, storing the values of Gamma is pointless, since we compute it.

If we go the route of changing to primitive reconstruction...a decent amount of work will be needed to update/test the schemes properly.  I think this would be a good step to implement, but It will take time.  

With all that, I see two options... 
1.) leave it as it is, the performance seems to be fine with it.  However, we can't say its validated...
2.) Take down the PR, make the primitive changes and make a new PR.

Im okay with either.

Reply:
I think option 2 is preferable but without wasting the work you've already done. Given that, and sSince this PR is basically almost finished (if not finished), why don't we go with 1+2?

Reply:
if you're ok with that, I give approval to this PR, we merge it, and jump to 2